clang/test/SemaSYCL/unique_stable_name.cpp

```diff
@@ -0,0 +1,223 @@
+  constexpr const char *F3_output = __builtin_unique_stable_name(F3);
+
+  // expected-error@#kernelSingleTask{{kernel instantiation changes the result of an evaluated '__builtin_unique_stable_name'}}
+  // expected-note@#kernel3func_call{{in instantiation of function template specialization}}
+  // expected-note@#USN_l7{{'__builtin_unique_stable_name' evaluated here}}
+  // expected-note@+1{{in instantiation of function template specialization}}
+  kernel_single_task<class kernel3>(F3);
+}
+
+// kernel4 - expect error
+// The current function is named with a lambda (i.e., takes a lambda as a
+// template parameter. Call the builtin on the current function
+// Then an empty lambda is passed to kernel
+// Error is thrown because the empty lambda is named based on kernel4func
+// Current function is a part of that name, causing its unique_stable_name
+// mangling to change
+template <typename Func>
+void kernel4func(const Func &F4) {
+  // expected-error@#kernelSingleTask{{kernel instantiation changes the result of an evaluated '__builtin_unique_stable_name'}}
+  // expected-note@#USN_F4{{'__builtin_unique_stable_name' evaluated here}}
+  constexpr const char *F4_output = __builtin_unique_stable_name(F4); // #USN_F4
+
+  // expected-error@#kernelSingleTask{{kernel instantiation changes the result of an evaluated '__builtin_unique_stable_name'}}
+  // expected-note@#kernel4func_call{{in instantiation of function template specialization}}
+  // expected-note@#USN_l7{{'__builtin_unique_stable_name' evaluated here}}
+  // expected-note@+1{{in instantiation of function template specialization}}
+  kernel_single_task<class kernel4>([]() {});
+}
+
+template <template <typename> typename Outer, typename Inner>
+struct S {
+  void operator()() const;
+};
+
+template <typename Ty>
+struct T {};
+
+template <typename Func>
+void kernel13_14func(const Func &F) {
+  // expected-error@#kernelSingleTask{{kernel instantiation changes the result of an evaluated '__builtin_unique_stable_name'}}
+  // expected-note@#USN_l11{{'__builtin_unique_stable_name' evaluated here}}
+  // expected-note@#USN_l12{{'__builtin_unique_stable_name' evaluated here}}
+  // expected-note@+1{{in instantiation of function template specialization}}
+  kernel_single_task<class kernel13>(F);
+  // expected-error@#kernelSingleTask{{kernel instantiation changes the result of an evaluated '__builtin_unique_stable_name'}}
+  kernel_single_task<class kernel14>(F);
+  // Using the same functor twice should be fine
+}
+
+template <typename Ty>
+auto func() -> decltype(__builtin_unique_stable_name(Ty::str));
+
+struct Derp {
+  static constexpr const char str[] = "derp derp derp";
+};
+
+template <typename T>
+static constexpr const char *output1 = __builtin_unique_stable_name(T);
+
+#define MACRO12()                \
+  auto l12 = []() { return 1; }; \
+  constexpr const char *l1_output = __builtin_unique_stable_name(l12); // #USN_l12
+// expected-note@-1{{expanded from macro 'MACRO12'}}
```

Comment:
Why do you have a note here all alone?  It should be with whatever the diagnostic stack is that causes the issue.